document_stores/elasticsearch/src/elasticsearch_haystack/document_store.py

```diff
@@ -115,8 +119,12 @@ def _search_documents(self, **kwargs) -> List[Document]:
                 from_=from_,
                 **kwargs,
             )
+
             documents.extend(self._deserialize_document(hit) for hit in res["hits"]["hits"])
-            from_ = len(documents)
+            from_ += len(documents)
+
+            if top_k and from_ >= top_k:
```

Comment:
Let's be explicit like we do above.

```suggestion
            if top_k is not None and from_ >= top_k:
```